internal/cli/log_streams_datadog.go

```diff
@@ -132,6 +132,16 @@ func updateLogStreamsDatadogCmd(cli *cli) *cobra.Command {
 				return fmt.Errorf("failed to read log stream with ID %s: %w", inputs.ID, err)
 			}
 
+			if oldLogStream.GetType() != string(logStreamTypeDatadog) {
+				return fmt.Errorf(
+					"the log stream with ID %q is of type %q instead of datadog, "+
+						"use 'auth0 logs streams update %s' to update it instead",
+					inputs.ID,
+					oldLogStream.GetType(),
+					oldLogStream.GetType(),
+				)
+			}
+
```

Comment:
Here we prevent a panic down below when we type cast the sink property, this safeguards us against passing a real log stream ID but of different type. This is replicated in all the update commands.